### PR TITLE
Fix attaching docstrings to where call syntax

### LINKED ISSUE #32960: Adding docstrings to signatures with type parameters not working
Adding docstrings to signatures with type parameters is not working properly:
```julia
julia> """asdsd"""
       sin(x::Vector{T}, ::T) where {T <: Integer}
ERROR: cannot document the following expression:

sin(x::Vector{T}, ::T) where T <: Integer
```
I have a feeling this should work.

### PR DESCRIPTION
**Pull request by mortenpi**
Fix #32960. I.e. that it would be possible to attach docstrings as follows:

```julia
"docstring"
sin(x::Vector{T}, ::T) where {T <: Integer}
```

---

The lookup (e.g. in `help>`) for the parametric methods is still a bit broken. E.g. if you have the following definitions:

```julia
"    tw(x::T) where T"
tw(x::T) where T = nothing
"    tw(x::T, y::U) where {T, U <: Integer}"
tw(x::T, y::U) where {T, U <: Integer} = nothing
"    tw(x::T, y::U, z::V) where T where U <: Integer where V <: AbstractFloat"
tw(x::T, y::U, z::V) where T where U <: Integer where V <: AbstractFloat = nothing
```

Looking up the single argument method does not work correctly for some reason:

```
help?> tw(0)
  tw(x::T) where T

  ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

  tw(x::T, y::U) where {T, U <: Integer}

  ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

  tw(x::T, y::U, z::V) where T where U <: Integer where V <: AbstractFloat

help?> tw(0,0)
  tw(x::T, y::U) where {T, U <: Integer}

help?> tw(0,0,0.0)
  tw(x::T, y::U, z::V) where T where U <: Integer where V <: AbstractFloat
```

But that is actually separate from the original issue.

### CODE REVIEW

**Comment by thofma** [comment]
Does this by accident also fix https://github.com/JuliaLang/julia/issues/29437?

The `Docs.signature` is (was?) really broken and added wrong signatures if the `where` is at the wrong place.

**Comment by mortenpi** [comment]
Anything holding this back? Should this be considered a bugfix or could this use a NEWS entry?

**Comment by StefanKarpinski** [comment]
Probably just a matter of who should review. Anyone particularly familiar with how the docstrings work that it would be helpful to have take a look at this? This seems worthy of a news entry since it seems a bit like a feature to me, but it could also be argued to be a bug fix.

**Review by fredrikekre — APPROVED** [review]
I guess this worked pre-`where`, e.g. https://github.com/JuliaLang/julia/pull/33446/files#diff-3f4eeb007d2f33142bda5be1dc90da75R154 maybe doesn't have to work anymore.

**Pull request merged**
